refactor(api): shares and likes

radurogojanumai opened the pull request.

radurogojanumai committed:
MSD-1089: refactor shares endpoint
add like / unlike for generator and sd

radurogojanumai committed:
ruff

radurogojanumai committed:
ruff 2

radurogojanumai committed:
fix conflicts

mplatzer commented on README.md:
```diff
@@ -124,8 +124,29 @@ mostly.share(g | sd | c, email)
 mostly.unshare(g | sd | c, email)
 
 g.shares()
+g.share(email)
```
let's just have a single line to demo syntax, i.e.
`g.share(email, "VIEW" | "EDIT" | "ADMIN")`

and drop the other 3 g.share() examples.

plus lets drop `mostly.share` and `mostly.unshare` from docs and from interface. this is the old deprecated interface.

mplatzer replied:
and please remove the `share_public` and `unshare_public` from the README, and comment out in the code. This is atm just for Admins, and we will want to rename that also in the Public API.

mplatzer commented on README.md:
```diff
@@ -124,8 +124,29 @@ mostly.share(g | sd | c, email)
+g.unlike()
+
 sd.shares()
+sd.share(email)
```
`sd.share(email, "VIEW" | "EDIT" | "ADMIN")`

mplatzer commented on README.md:
```diff
@@ -124,8 +124,29 @@ mostly.share(g | sd | c, email)
+
 c.shares()
+c.share(email)
+c.share(email, "EDIT")
```
`c.share(email, "VIEW" | "EDIT" | "ADMIN")`

and drop `c.share(email)`

mplatzer commented on tools/model.py:
```diff
@@ -257,7 +351,54 @@ def data(
+
+    def unshare(self, user_email: str):
+        """
+        Stop sharing the Synthetic Data with the specified user
```
use `synthetic dataset` consistently instead of `Synthetic Data` throughout the code

michdr commented on mostlyai/model.py:
```diff
@@ -353,7 +353,54 @@ def config(self) -> dict[str, Any]:
+        """
+        Retrieve list of shares
+
+        :return: list of ResourceShares
```
Just `ResourceShares`, which contains a list inside of that object.

michdr replied:
Same goes with the other identical method signatures

michdr reviewed (approved):
A couple of minor corrections in docstrings. Other than that, looks good 👍🏼

radurogojanumai committed:
test

radurogojanumai committed:
removing share_public(), unshare_public() from Connector, Generator, SyntheticDataset
removing mostly.share() and mostly.unshare()
type hints on return of shares()
consistent comments, eg synthetic dataset, ResourceShares
cleanup of README.md
regenerated the model.py

radurogojanumai committed:
ruff

mplatzer committed:
Update README.md

zachtil commented on README.md:
```diff
@@ -117,15 +117,24 @@ sp = mostly.probe(g, size=size)
-mostly.share(g | sd | c, email)
-mostly.unshare(g | sd | c, email)
+c.shares()
+c.share(email, "VIEW" | ""EDIT" | "ADMIN")
```
Duplicate double-quote

```suggestion
c.share(email, "VIEW" | "EDIT" | "ADMIN")
```

zachtil commented on README.md:
```diff
@@ -117,15 +117,24 @@ sp = mostly.probe(g, size=size)
+c.unshare(email)
 
 g.shares()
+g.share(email, "VIEW" | ""EDIT" | "ADMIN")
```
Duplicate double-quote

```suggestion
g.share(email, "VIEW" | "EDIT" | "ADMIN")
```

zachtil commented on README.md:
```diff
@@ -117,15 +117,24 @@ sp = mostly.probe(g, size=size)
+
 sd.shares()
-c.shares()
+sd.share(email, "VIEW" | ""EDIT" | "ADMIN")
```
Duplicate double-quote

```suggestion
sd.share(email, "VIEW" | "EDIT" | "ADMIN")
```
